CONTRIBUTING.md

```diff
@@ -0,0 +1,194 @@
+4. Label the issue with "ready" (to be worked on).
+5. A developer "claims" the issue (assigns themselves), create a new branch, begins work on implementation.
+
+### Implementation Work
+
+Requirements:
+- Spark component is a wrapper around, or is based on, a Material-UI component(s).
+  - Imports from MUI should be named and prefixed with "Mui"
+```tsx
+// FILE: src/Button.tsx
+import { 
+  ButtonBase as MuiButtonBase,
+  ButtonBaseProps as MuiButtonBaseProps,
+} from '@material-ui/core';
+``` 
+- Spark component props are defined as a TypeScript interface and extend the underlying component's props.
+```tsx
+// FILE: src/Button.tsx
+export interface ButtonProps 
+  extends Omit<MuiButtonBaseProps, 'unwanted_prop'> {
+    custom_prop?: 'value';
+  }
+```
+- Styling is done using styled-components or emotion
+- Component is functional React component.
+- Component name is prefixed with "Spark", but exported as proper name.
+```tsx
+// FILE: src/Button.tsx
+const SparkButton: FC<ButtonProps> = (props) => {...}
+export { SparkButton as Button };
+```
+- One component per file
+  - internal (sub-)components can be moved to `src/internal/` as needed.
+  - components with composition patterns that customize other existing Spark components should also be prefixed accordingly 
+```tsx
+// FILE: src/Button.tsx
+import { Button, ButtonProps } from './Button';
+export interface DropdownButtonProps = ButtonProps
+const SparkDropdownButton: React.FC<DropdownButtonProps> = props => {
+  const { id, handleClick } = React.useContext(Context);
+  return (
+    <Button
+      aria-controls={id}
+      aria-haspopup="true"
+      onClick={handleClick}
+      {...props}
+    />
+  );
+};
+export {
+  SparkDropdownButton as DropdownButton,
+};
+```
+- Component and Props interface are exported in `src/index.ts`.
+
+### Other
+
+If somebody claims an issue but doesn't follow up for more than a week, then it's fine to take it over -- you should leave a comment to let others know.
+
+If there has been no activity on an issue for 7 to 14 days, it is safe to assume that nobody is working on it.
+
+## Keep a Pull Request small
+
+When in doubt, keep your Pull Requests small. To give a Pull Request the best chance of getting accepted, don't bundle more than one feature or bug fix per Pull Request. It's often best to create two smaller Pull Requests than one big one.
+
+### Open an Issue first
+
+Before working on any change besides chores, it is best to open an issue first and discuss it with the maintainers.
+
+## Get Up and Running
+
+There will be differences depending on if you are inside or outside of Prenda. See [Outside of Prenda?](##-outside-of-prenda) for our outlook on this.
+
+1. Inside of Prenda? Skip this step. Outside of Prenda? Fork the repository.
+2. Inside of Prenda? Clone the repository to your local machine. Outside of Prenda? Clone the fork to your local machine, add upstream remote, and synchronize your local with upstream. Else: goto 3. I
```

Comment:
This part is a bit confusing. I think we could simply go right to the headings `# Inside of Prenda` then list all steps in order followed by `# Outside of Prenda` with all steps. These could both have a `goto 3` at the end.

Reply:
Fair enough, I was trying to cut down on repetition, but i could tell it was getting confusing.

Reply:
Pushed a change for this